### ISSUE TITLE
Api changes in 1.4

### ISSUE THREAD

**Issue by aaren** [NanoComp/harminv#2]
9f948988 has changed the api. (Now released with version 1.4)

The arguments of `harminv_get_omega` and `harminv_get_amplitude` have changed to require a `harminv_complex` as first argument.

Was this change necessary? Do you expect it to be permanent?

I can update my python interface ([aaren/pharminv](http://github.com/aaren/pharminv)) to match the change without much problem, but I'm not sure whether you intended this change or not.

(harminv 1.3.1 is distributed with Debian Jessie)

**Comment by stevengj** [comment]
Yes, the API change is documented in the NEWS and is expected to be permanent.  It was necessary in order to make the C99 and C++ ABIs (binary interfaces) compatible on non-x86 platforms (i.e. so that both C++ and C99 code can link to the same library).

**Comment by stevengj** [comment]
See the discussion in stevengj/meep#13

**Comment by stevengj** [comment]
You can maintain source compatibility with older harminv versions in your code via:

``` c
#if HARMINV_VERSION_MAJOR < 1 || (HARMINV_VERSION_MAJOR == 1 && HARMINV_VERSION_MINOR < 4)
#  define harminv_get_amplitude(pa, d, k) *(pa) = harminv_get_amplitude(d, k)
#  define harminv_get_omega(pw, d, k) *(pw) = harminv_get_omega(d, k)
#endif
```

as in stevengj/meep@367115c37893a822697d60439c14c952dbfc12dc

**Comment by aaren** [comment]
Thanks for the explanation. I've fixed it in aaren/pharminv@be410c2.

I've released [pharminv 0.3](https://pypi.python.org/pypi/pharminv/0.3) with this fix.
